spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmount.java

```diff
@@ -0,0 +1,307 @@
+		Assert.notNull(predicate, () -> "Predicate must not be null");
+		if (predicate.test(this)) {
+			return Optional.of(this);
+		}
+		return Optional.empty();
+	}
+
+	/**
+	 * Apply the provided mapping function to this.
+	 * @param <T> the type of the result of the mapping function
+	 * @param mapper a mapping function to apply to the value.
+	 * @return the result of applying a mapping function
+	 */
+	public <T> T map(@NonNull Function<? super DigitalAmount, ? extends T> mapper) {
+		Assert.notNull(mapper, () -> "Mapper must not be null");
+		return mapper.apply(this);
+	}
+
+	/**
+	 * Returns a {@code DigitalAmount} whose value is {@code (this +
+	 * other)}.
+	 * @param amount value to be added to this {@code DigitalAmount}.
+	 * @param unit {@code DigitalUnit} of this amount.
+	 * @return {@code this + other}
+	 */
+	public DigitalAmount add(long amount, @NonNull DigitalUnit unit) {
+		return add(fromUnit(amount, unit));
+	}
+
+	/**
+	 * Returns a {@code DigitalAmount} whose value is {@code (this -
+	 * other)}.
+	 * @param amount value to be subtracted to this {@code DigitalAmount}.
+	 * @param unit {@code DigitalUnit} of this amount.
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(long amount, @NonNull DigitalUnit unit) {
+		return subtract(fromUnit(amount, unit));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this +
+	 * other)}.
+	 * @param other value to be added to this {@code DigitalAmount}.
+	 * @return {@code this + other}
+	 */
+	public DigitalAmount add(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return new DigitalAmount(Math.addExact(this.amount, other.amount));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this -
+	 * other)}.
+	 * @param other value to be subtracted from this {@code DigitalAmount}.
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return new DigitalAmount(Math.subtractExact(this.amount, other.amount));
+	}
+
+	@Override
+	public int compareTo(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return Long.compare(this.amount, other.amount);
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		DigitalAmount that = (DigitalAmount) o;
+		return this.amount == that.amount;
+	}
+
+	@Override
+	public int hashCode() {
+		return Long.hashCode(this.amount);
+	}
+
+	/**
+	 * Print this {@code DigitalAmount} using the given unit.
+	 * @param unit the {@code DigitalUnit} to use if the value doesn't specify,
+	 * {@link DigitalUnit#BYTES} will be used
+	 * @return the printed result
+	 * @see DigitalAmountStyle
+	 */
+	public String print(@Nullable DigitalUnit unit) {
```

Comment:
We don't use `@Nullable` in Spring Boot yet and I am not keen to introduce it at this point. 

This `print` method should go away in benefit of `toString`